lib/drivers/direct/index.js

```diff
@@ -147,9 +156,31 @@ Driver.prototype._containerById = function(instanceId) {
       server: this._server,
       instanceId: instanceId,
     });
+
     container.instance = {
       driverMeta: {},
     };
+
+    if (this._app) {
+      var port = mandatory(this._server.port());
+      var channel = this._wsRouter.createChannel(onRequest);
+      var request = channel.request.bind(channel);
+      var token = channel.getToken();
+      var control = util.format('ws://%s@127.0.0.1:%d', token, port);
+
+      debug('container control socket: %s', control);
+
+      // Patch our request method in... because the container calls request
+      // itself (which should be changed sometime).
+      container.request = request;
+      container.channel = channel;
+      container.setStartOptions({control: control});
+
+      function onRequest(req, callback) {
```

Comment:
Tests fail for me locally because this isn't legal in `'strict mode';` apparently. I'm not sure why eslint doesn't detect it or how CI's tests are passing with it like this.. I'll include a fix in #228.